### ISSUE TITLE
Is it worthwhile to avoid requiring collision resistance from our hash?

### ISSUE THREAD

**Issue by bwesterb** [davidben/merkle-tree-certs#45]

**Comment by davidben** [comment]
With a CA-provided randomizer, we'd still rely on second preimage resistance for transparency, right? (I.e. we couldn't halve the hash size.) Though I suppose we wouldn't rely on it for authenticity.

**Comment by bwesterb** [comment]
Yes, you're right: even with the CA-provided randomiser, we would still rely on *collision resistance* to prevent the CA from providing two world views. Indeed, if we'd truncate hashes to 16 bytes, then with 2^64 work, a CA could find two different batches with the same root hash.

I guess we couldn't get rid of that without significant modifications (see #48). Still I think it might be worthwhile to reduce reliance of collision resistance.

**Comment by davidben** [comment]
Post-Photosynthesis triage: eh, I guess we've decided to just import a CT/tlog-style tree structure. Probably can close this.

**Comment by bwesterb** [comment]
Let's [reopen](https://mailarchive.ietf.org/arch/msg/plants/FN8L2-C2AjN7QxpX6RC7b3tnxi8/) this.
